docs/ossm/injection/README.md

```diff
@@ -0,0 +1,124 @@
+
+In most cases, there will be a single `IstioRevision` resource per `Istio` resource. During a revision based upgrade, there may be multiple `IstioRevision` instances present, each representing an independent control plane. 
+
+The available revision names can be checked with the command:
+
+```console
+$ oc get istiorevision
+NAME              READY   STATUS    IN USE   VERSION   AGE
+my-mesh-v1-23-0   True    Healthy   False    v1.23.0   114s
+```
+
+### Enabling sidecar injection - "default" revision
+
+When the service mesh's `IstioRevision` name is "default", it's possible to use following labels on a namespace or a pod to enable sidecar injection:
+| Resource | Label | Enabled value | Disabled value |
+| --- | --- | --- | --- |
+| Namespace | `istio-injection` | `enabled` | `disabled` |
+| Pod | `sidecar.istio.io/inject` | `"true"` | `"false"` |
+
+### Enabling sidecar injection - other revisions
+
+When the `IstioRevision` name is not "default", then the specific `IstioRevision` name must be used with the `istio.io/rev` label to map the pod to the desired control plane while enabling sidecar injection. 
+
+For example, with the revision shown above, the following labels would enable sidecar injection:
+| Resource | Enabled Label | Disabled Label |
+| --- | --- | --- |
+| Namespace | `istio.io/rev=my-mesh-v1-23-0` | `istio-injection=disabled` |
+| Pod | `istio.io/rev=my-mesh-v1-23-0` | `sidecar.istio.io/inject="false"` |
+
+> [!NOTE]
+> Next tech preview release will add support for revision tags which will simplify the upgrade process significantly and also will allow to use the `istio-injection` label for cases when `IstioRevision` name is not "default".
+
+### Sidecar injection logic
+
+If the `istio-injection` label and the `istio.io/rev` label are both present on the same namespace, the `istio-injection` label (mapping to the "default" revision) will take precedence.
+
+The injector is configured with the following logic:
+
+1. If either label (`istio-injection` or `sidecar.istio.io/inject`) is disabled, the pod is not injected.
+2. If either label (`istio-injection` or `sidecar.istio.io/inject` or `istio.io/rev`) is enabled, the pod is injected.
+
+### Example: Enabling sidecar injection
+Prerequisites:
+- The OpenShift Service Mesh operator has been installed
+- An Istio CNI resource has been created
+
+1. Prepare `default` `istio.yaml`:
+    ```yaml
+    kind: Istio
+    apiVersion: sailoperator.io/v1alpha1
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: InPlace
+      version: v1.23.0
+    ```
+1. Create the `default` Istio CR in `istio-system` namespace:
+    ```bash
+    oc apply -f istio.yaml
+    ```
```

Comment:
Maybe add a `oc wait` here for the `Istio` resource to be ready? Otherwise users won't catch that something happened with their `Istio` install and will wonder why the pod wasn't getting injected.

Reply:
Added, thank you.